generators/pkg/generator/generate.go

```diff
@@ -3,20 +3,40 @@
 package generator
 
 import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
 	v1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/yaml"
 )
 
+// GitHub defines Github fields
+type GitHub struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              GitHubSpec `json:"spec"`
+}
+
 // GithubSpec defines Github spec
 type GitHubSpec struct {
 	URL   string         `json:"url,omitempty"`
 	Steps []v1beta1.Step `json:"steps,omitempty"`
 }
 
```

Comment:
we can put the two consts in a block:
```
const(
 GitCLoneURL = ...
 GitHubStatusURL = ...
)
```

Reply:
I don't think we should fetch master, since this may result in unintended/inconsistent behavior. We should rely on the on cluster task for versioning for the time being.

Reply:
Is it a good way to fix the commitId for the gitClone task? 
e.g.
```
https://raw.githubusercontent.com/tektoncd/catalog/2fa2e40357bfebbd739d475eab233972a8e16782/git/git-clone.yaml
```

Reply:
> If we are going with the cluster task approach, let's make sure we document somewhere that the user has to install a few tasks themselves.

Is it a good way to add instructions in the README file?

Reply:
Yup! That's probably the best spot for it.